cubids/cli.py

```diff
@@ -121,44 +136,115 @@ def cubids_validate():
-                        shutil.copy2(fi, output)
-
-                    # run the validator
-                    nifti_head = opts.ignore_nifti_headers
-                    subj_consist = opts.ignore_subject_consistency
-                    call = build_validator_call(tmpdirname,
-                                                nifti_head,
-                                                subj_consist)
+                # TMPDIR isn't networked (available on login + exec nodes), so use bids_dir
+                tmpdir = tempfile.TemporaryDirectory(dir=opts.bids_dir, prefix=".")
+                for fi in files_list:
+
+                    # cut the path down to the subject label
+                    bids_start = fi.find(subject)
+
+                    # maybe it's a single file
+                    if bids_start < 1:
+                        bids_folder = tmpdir.name
+                        fi_tmpdir = tmpdir.name
+
+                    else:
+                        bids_folder = Path(fi[bids_start:]).parent
+                        fi_tmpdir = tmpdir.name + '/' + str(bids_folder)
+
+                    if not os.path.exists(fi_tmpdir):
+                        os.makedirs(fi_tmpdir)
+                    output = fi_tmpdir + '/' + str(Path(fi).name)
+                    shutil.copy2(fi, output)
+
+                # run the validator
+                nifti_head = opts.ignore_nifti_headers
+                subj_consist = opts.ignore_subject_consistency
+                call = build_validator_call(tmpdir.name,
+                                            nifti_head,
+                                            subj_consist)
+
+                if queue is False:
                     ret = run_validator(call)
-                    # parse output
+                    # execute and parse output immediately
                     if ret.returncode != 0:
                         logger.error("Errors returned "
-                                     "from validator run, parsing now")
+                                        "from validator run, parsing now")
 
                     # parse the output and add to list if it returns a df
                     decoded = ret.stdout.decode('UTF-8')
                     tmp_parse = parse_validator_output(decoded)
                     if tmp_parse.shape[1] > 1:
                         tmp_parse['subject'] = subject
                         parsed.append(tmp_parse)
+                else:
+                    queue.append({
+                        'call': call,
+                        'tmpdir': tmpdir
+                    })
+
+
+            if opts.drmaa:
+                try:
+                    drmaa_ses = drmaa.Session()
+                    drmaa_ses.initialize()
+                    tmpfiles = []
+                    jids = []
+
+                    for batch in build_drmaa_batch(queue):
+                        tmp = tempfile.NamedTemporaryFile(delete=False, dir=opts.bids_dir, prefix=".", suffix=".sh")
```

Comment:
Is this something a user can customize? Or will they need to customize it? does this work out of the box on cubic?

Reply:
Not sure what would need to be customized? It indeed works out of the box on CUBIC, LSF also supports DRMAA but PMACS set it up in a weird way and sounded uninteresting in changing that when I asked :(